File: jcv-core/src/main/java/com/ekino/oss/jcv/core/comparator/RegexComparator.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator;

import java.util.regex.Pattern;

import com.ekino.oss.jcv.core.JsonValueComparator;

import lombok.RequiredArgsConstructor;
import org.skyscreamer.jsonassert.ValueMatcherException;

@RequiredArgsConstructor
public class RegexComparator implements JsonValueComparator<String> {

    private final Pattern pattern;

    @Override
    public boolean hasCorrectValue(String actual, String expected) {
        if (actual != null && pattern.matcher(actual).matches()) {
            return true;
        }
        throw new ValueMatcherException("Value does not match pattern /" + pattern + "/", expected, actual);
    }
}

File: jcv-core/src/main/kotlin/com/ekino/oss/jcv/core/comparator/RegexComparator.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator

import com.ekino.oss.jcv.core.JsonValueComparator
import org.skyscreamer.jsonassert.ValueMatcherException
import java.util.regex.Pattern

class RegexComparator(private val pattern: Pattern) : JsonValueComparator<String> {

  override fun hasCorrectValue(actual: String?, expected: String?): Boolean {
    if (actual != null && pattern.matcher(actual).matches()) {
      return true
    }
    throw ValueMatcherException("Value does not match pattern /$pattern/", expected, actual)
  }
}
